language/bytecode-verifier/src/dependencies.rs

```diff
@@ -338,3 +338,46 @@ impl<'a> DependencyChecker<'a> {
         }
     }
 }
+
+pub struct CyclicModuleDependencyChecker {}
+
+impl CyclicModuleDependencyChecker {
+    pub fn verify_module<F>(module: &CompiledModule, dep_loader: F) -> VMResult<()>
```

Comment:
Naming this thing "dep_loader" feels a bit leaky as this bytecode verifier knows nothing about the loader. 

Perhaps a more declarative name, something like "module_dependencies"?

Reply:
How about, `immediate_module_dependencies`?

Reply:
Works for me!